icetea_lib/TestSuite/TestcaseContainer.py

```diff
@@ -370,7 +370,11 @@ def run(self, forceflash=False):
 
         stop_time = datetime.datetime.now()
 
-        result = tc_instance.get_result(tc_file=self._filepath)
+        if tc_instance.results:
```

Comment:
isn't there any way to avoid this condition and simplify code.. ?

Reply:
Well, there's always options, but they just tend to move the code somewhere else.

Reply:
if test has only resultList even it contains only one element, doesn’t then be more simple without extra conditions.🙄...?

Reply:
That would mean reworking some other logic elsewhere as well. The ResultList as result is an optional feature and most test cases don't do it at the moment. It's something that we might rework at some point.

Reply:
agree. I propose to change that in some point to return ResultList only and always, then there is no needs for additional logic depends on how many results Test returns....

Reply:
Yes, we'd need to rework the way the test case returns it's result so that the API instead appends a result into a ResultList which then gets returned when TestSuite asks for the result.